merlion/models/utils/rolling_window_dataset.py

```diff
@@ -1,46 +1,89 @@
+        :param shuffle: True or False to randomly shuffle the returning window, please note that the
+            time series data itself will never be shuffled
+        :param ts_index: keep original TimeSeries internally for all the slicing, and output TimeSeries.
+            by default, Numpy array will handle the internal data workflow and Numpy array will be the output.
+        :param batch_size: if >= 1: will yield a list of (past, future) with batch_size
+            if None: will give a one-shot dataset for all the rolling windows.
+        """
+        assert isinstance(data, (TimeSeries, pd.DataFrame)), \
+            "RollingWindowDataset expects to receive TimeSeries or pd.DataFrame data "
+        if isinstance(data, TimeSeries):
+            data = data.align()
+            self.dim = data.dim
+        else:
+            assert isinstance(data.index, pd.DatetimeIndex)
+            assert ts_index is False, "Only TimeSeries data support ts_index = True "
+            self.dim = data.shape[1]
+
         self.batch_size = batch_size
         self.ts_index = ts_index
 
+        self.target_seq_index = target_seq_index
+        if self.target_seq_index is None:
+            logger.info(
+                f"target_seq_index is None, therefore, the future data will be rolling along "
+                f"the entire sequence dimension with timestamp increment by 1, i.e, self.n_future = 1."
+                f"This is the rolling strategy for autoregression algorithm where all the future "
+                f"sequences need to be provided as the training prior. "
+                f"If you are not expecting this behavior, please properly set up the target_seq_index "
+                f"as a valid integer. "
+            )
+            # label rolling along the sequence dimension
+            self._label_axis = 1
+            self.n_future = 1
+        else:
+            # label rolling along the time dimension
+            self._label_axis = 0
+            self.n_future = n_future
+
+        self.n_past = n_past
+
         if ts_index:
             self.data_ts = data
-            self.target_ts = data.univariates[data.names[target_seq_index]]
+            self.target_ts = data.univariates[data.names[target_seq_index]] if self._label_axis == 0 else None
         else:
-            data_df = data.to_pd()
+            data_df = data.to_pd() if isinstance(data, TimeSeries) else data
             self.data = data_df.values
             self.timestamp = data_df.index
-            self.target = data.univariates[data.names[target_seq_index]].values
-            self.target_timestamp = data.univariates[data.names[target_seq_index]].index
-
-        self.target_seq_index = target_seq_index
-        self.maxlags = maxlags
-        self.forecast_steps = forecast_steps
+            self.target = data_df[data_df.columns[target_seq_index]].values if self._label_axis == 0 else None
```

Comment:
`data_df.iloc[:, target_seq_index]` is more concise

Reply:
Done